src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -870,9 +870,14 @@ def filter_and_paginate(
         # Sorting
         column, operand = filter_model.sorting_params
         if operand == SorterOps.DESCENDING:
-            query = query.order_by(desc(getattr(table, column)))
+            sort_clause = desc(getattr(table, column))
         else:
-            query = query.order_by(asc(getattr(table, column)))
+            sort_clause = asc(getattr(table, column))
+
+        # We always add the `id` column as a tiebreaker to ensure a stable,
+        # repeatable order of items, otherwise subsequent pages might contain
+        # the same items.
+        query = query.order_by(sort_clause, asc(table.id))
```

Comment:
are we sure here that every entity in the db has an id?

Reply:
Not every entity in the DB has an ID, no. However, all entities supported by the generic filter and pagination feature inherit from `BaseSchema` and they all have an ID column.